### DISCUSSION TITLE
How to build a x64 version of an app using WiX Toolset

### DISCUSSION THREAD

**Discussion by jforward5** [wixtoolset/issues#8041] [Questions]
I've been trying to get my WiX project to build my app as an x64 version but it always tries to build it as "AnyCPU".  

I'm trying to use the preprocessor variables in my wixproj to build the correct arch, like this:
`<Exec Command="dotnet publish &quot;..\ATLAS Agent 2.0\ATLAS Agent 2.0.csproj&quot; -c $(Configuration) -a $(Platform) -o &quot;..\ATLAS Agent 2.0\bin\$(Configuration)\net7.0-windows\win-$(Platform)\publish&quot; --self-contained" />`

Then from the command line I do this:
`dotnet build ".\ATLAS Agent 2.0 Setup\ATLAS Agent 2.0 Setup.wixproj" -c Release -a x64 --self-contained`

And I would expect the RuntimeIdentifier to be set to "win-x64", but instead I am getting this error:
`C:\Program Files\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targe
ts(90,5): error NETSDK1083: The specified RuntimeIdentifier 'win-AnyCPU' is not recognized. See https://aka.ms/netsdk10
83 for more information. [C:\Users\Joel.Brecker\source\repos\atlas-agent-2.0\ATLAS Agent 2.0\ATLAS Agent 2.0.csproj] [C
:\Users\Joel.Brecker\source\repos\atlas-agent-2.0\ATLAS Agent 2.0 Setup\ATLAS Agent 2.0 Setup.wixproj]
C:\Users\Joel.Brecker\source\repos\atlas-agent-2.0\ATLAS Agent 2.0 Setup\ATLAS Agent 2.0 Setup.wixproj(57,5): error MSB
3073: The command "dotnet publish "..\ATLAS Agent 2.0\ATLAS Agent 2.0.csproj" -c Release -a AnyCPU -o "..\ATLAS Agent 2
.0\bin\Release\net7.0-windows\win-AnyCPU\publish" --self-contained" exited with code 1.`

My question is, why is it not recognizing the "architecture" that I am passing in the build command?  What am I missing?

I've attached my "wixproj" file in case it helps.
[wixproj.txt](https://github.com/wixtoolset/issues/files/14575117/wixproj.txt)

[EDIT]
By hard coding the arch into the command in the wixproj file I can get it to build x64, but then it tells me this for every component:
`ICE80: This 32BitComponent ATLASMgmtSvrService uses 64BitDirectory MgmtSvrFolder`

Not sure why it is still trying to compile 32bit.  Is there some way to make Heat use 64bit?
If i change the folder to "ProgramFiles6432Folder" then it installs the app as 32 bit.

An added bit of information:

- These errors don't happen when I build in Visual Studios
- These errors are only present when building from command line...
- After building in VS it does install as 64 bit

**Comment by jforward5** [comment]
I found the solution after following a trail of other users who had similar issues.  Basically you have to add the following to your wixproj file:
`<InstallerPlatform Condition=" '$(Platform)' == '' ">x64</InstallerPlatform>`

And it must live inside one of the "PropertyGroup"s

**Reply by robmen** [reply]
Usually, you just use `Platform`, not the internal property `InstallerPlatform`. You _can_ use `InstallerPlatform` but should understand the difference.

**Comment by joelfrancis1122** [comment]
whats that
